File: bridge/python/src/destack/_impl/mir/node.py
from __future__ import annotations


def local_node_id_matches(left, right) -> bool:
    """Return whether two local node ids refer to the same MIR node."""

    return left.id == right.id


def get_local_node_value(mapping, key):
    """Return one map value keyed by local node id."""

    return mapping.get(key)

File: bridge/typescript/src/_impl/mir/node.ts
import type { LocalNodeId } from "../../_generated/mir/tree/node.js";

/** Return whether two local node ids refer to the same MIR node. */
export function localNodeIdMatches(left: LocalNodeId, right: LocalNodeId): boolean {
    return left.id === right.id;
}

/** Return one map value keyed by local node id. */
export function getLocalNodeValue<Value>(
    map: ReadonlyMap<LocalNodeId, Value>,
    key: LocalNodeId,
): Value | undefined {
    for (const [candidate, value] of map) {
        // compare structural node ids because JS maps key objects by identity
        if (localNodeIdMatches(candidate, key)) {
            return value;
        }
    }

    return undefined;
}
